// src/pages/Signup.jsx
// src/pages/Signup.jsx (With Email/Username/Password)
import React, { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { Scan, ChevronLeft, ChevronRight, Check, Sparkles, User, Utensils, Heart, Ruler, Weight, Mail, Lock, Eye, EyeOff } from 'lucide-react';
import { Input } from '@/components/ui/input';
import { Button } from '@/components/ui/button';
import { Label } from '@/components/ui/label';
import { Progress } from '@/components/ui/progress';
import { Badge } from '@/components/ui/badge';
import { Card, CardContent } from '@/components/ui/card';
import { Alert, AlertDescription } from '@/components/ui/alert';
import { APP_NAME, DIET_OPTIONS, ALLERGEN_OPTIONS, HEALTH_FOCUS_OPTIONS, GENDER_OPTIONS } from '../lib/constants';
import { useUser } from '../context/UserContext';

const Signup = () => {
  const navigate = useNavigate();
  const { login } = useUser();
  const [currentSlide, setCurrentSlide] = useState(0);
  const [showPassword, setShowPassword] = useState(false);
  const [showConfirmPassword, setShowConfirmPassword] = useState(false);
  const [error, setError] = useState('');
  
  const [formData, setFormData] = useState({
    email: '',
    username: '',
    password: '',
    confirmPassword: '',
    name: '',
    //gender: '',
    age: '',
    dietPreference: '',
    allergens: [],
    healthFocus: [],
    height: '',
    weight: ''
  });

  const totalSlides = 2;
  const progress = (currentSlide / totalSlides) * 100;

  const handleInputChange = (e) => {
    setFormData({
      ...formData,
      [e.target.name]: e.target.value
    });
    setError('');
  };

  const handleDietSelect = (diet) => {
    setFormData({ ...formData, dietPreference: diet });
  };

  const handleAllergenToggle = (allergen) => {
    setFormData(prev => ({
      ...prev,
      allergens: prev.allergens.includes(allergen)
        ? prev.allergens.filter(a => a !== allergen)
        : [...prev.allergens, allergen]
    }));
  };

  const handleHealthFocusToggle = (focus) => {
    setFormData(prev => ({
      ...prev,
      healthFocus: prev.healthFocus.includes(focus)
        ? prev.healthFocus.filter(f => f !== focus)
        : [...prev.healthFocus, focus]
    }));
  };

  const validateSlide0 = () => {
    if (!formData.email.trim()) {
      setError('Email is required');
      return false;
    }
    if (!/^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(formData.email)) {
      setError('Please enter a valid email address');
      return false;
    }
    if (!formData.username.trim()) {
      setError('Username is required');
      return false;
    }
    if (formData.username.length < 3) {
      setError('Username must be at least 3 characters');
      return false;
    }
    if (!formData.password) {
      setError('Password is required');
      return false;
    }
    if (formData.password.length < 6) {
      setError('Password must be at least 6 characters');
      return false;
    }
    if (formData.password !== formData.confirmPassword) {
      setError('Passwords do not match');
      return false;
    }

    // Check if email or username already exists
    const existingUsers = JSON.parse(localStorage.getItem('nutriscan_users') || '[]');
    if (existingUsers.some(u => u.email === formData.email)) {
      setError('Email already registered. Please login.');
      return false;
    }
    if (existingUsers.some(u => u.username === formData.username)) {
      setError('Username already taken. Please choose another.');
      return false;
    }

    return true;
  };

  const validateSlide1 = () => {
    if (!formData.name.trim()) {
      setError('Name is required');
      return false;
    }
    // if (!formData.gender) {
    //   setError('Please select your gender');
    //   return false;
    // }
    return true;
  };

  const handleNext = () => {
    setError('');

    if (currentSlide === 0 && !validateSlide0()) {
      return;
    }
    if (currentSlide === 1 && !validateSlide1()) {
      return;
    }

    if (currentSlide < totalSlides) {
      setCurrentSlide(currentSlide + 1);
    }
  };

  const handlePrevious = () => {
    if (currentSlide > 0) {
      setCurrentSlide(currentSlide - 1);
      setError('');
    }
  };

  const handleSkip = () => {
    if (currentSlide === totalSlides) {
      handleSubmit();
    } else {
      setCurrentSlide(currentSlide + 1);
    }
  };

  const handleSubmit = () => {
    if (currentSlide === 0 && !validateSlide0()) {
      return;
    }
    if (currentSlide === 1 && !validateSlide1()) {
      return;
    }

    // Save user credentials to localStorage
    const existingUsers = JSON.parse(localStorage.getItem('nutriscan_users') || '[]');
    const newUser = {
      email: formData.email,
      username: formData.username,
      password: formData.password, // In production, this should be hashed
      name: formData.name,
      gender: formData.gender,
      age: formData.age,
      dietPreference: formData.dietPreference,
      allergens: formData.allergens,
      healthFocus: formData.healthFocus,
      height: formData.height,
      weight: formData.weight,
      createdAt: new Date().toISOString()
    };

    existingUsers.push(newUser);
    localStorage.setItem('nutriscan_users', JSON.stringify(existingUsers));

    // Login user
    login(newUser);
    navigate('/dashboard');
  };

  const slideVariants = {
    0: { icon: Mail, title: "Create Account", subtitle: "Set up your login credentials" },
    1: { icon: User, title: "Basic Information", subtitle: "Let's get to know you" },
    2: { icon: Utensils, title: "Dietary Preferences", subtitle: "Customize your nutrition" },
    3: { icon: Heart, title: "Health & Wellness", subtitle: "Your fitness goals" }
  };

  const currentSlideInfo = slideVariants[currentSlide];
  const SlideIcon = currentSlideInfo.icon;

  return (
    <div className="min-h-screen bg-gradient-to-br from-emerald-50 via-white to-teal-50 flex items-center justify-center p-4 relative overflow-hidden">
      {/* Animated background */}
      <div className="absolute top-20 left-10 w-96 h-96 bg-emerald-200 rounded-full mix-blend-multiply filter blur-3xl opacity-20 animate-pulse" />
      <div className="absolute bottom-20 right-10 w-96 h-96 bg-teal-200 rounded-full mix-blend-multiply filter blur-3xl opacity-20 animate-pulse" style={{ animationDelay: '1.5s' }} />
      
      <div className="w-full max-w-3xl relative z-10">
        {/* Header */}
        <div className="text-center mb-8">
          <div className="inline-flex items-center gap-2 mb-4">
            <div className="relative">
              <Scan className="w-12 h-12 text-emerald-600" />
              <Sparkles className="w-5 h-5 text-teal-500 absolute -top-1 -right-1 animate-pulse" />
            </div>
            <span className="text-3xl font-bold bg-gradient-to-r from-emerald-600 via-teal-600 to-cyan-600 bg-clip-text text-transparent">
              {APP_NAME}
            </span>
          </div>
          <h1 className="text-3xl font-bold text-gray-900 mb-2">Create Your Account</h1>
          <p className="text-gray-600">Personalize your nutrition journey in 4 simple steps</p>
        </div>

        {/* Progress Bar */}
        <div className="mb-8">
          <Progress value={progress} className="h-2" />
          <div className="flex justify-between mt-2 text-sm text-gray-600">
            <span>Step {currentSlide + 1} of {totalSlides + 1}</span>
            <span>{Math.round(progress)}% Complete</span>
          </div>
        </div>

        {/* Slide Indicator Dots */}
        <div className="flex justify-center items-center gap-3 mb-8">
          {[0, 1, 2, 3].map((step) => (
            <div key={step} className="flex items-center">
              <div className={`flex items-center justify-center w-10 h-10 rounded-full font-semibold transition-all duration-300 ${
                currentSlide >= step
                  ? 'bg-gradient-to-r from-emerald-600 to-teal-600 text-white shadow-lg scale-110'
                  : 'bg-gray-200 text-gray-500'
              }`}>
                {currentSlide > step ? <Check className="w-5 h-5" /> : step + 1}
              </div>
              {step < 3 && (
                <div className={`w-16 h-1 mx-2 rounded transition-all duration-300 ${
                  currentSlide > step ? 'bg-gradient-to-r from-emerald-600 to-teal-600' : 'bg-gray-200'
                }`}></div>
              )}
            </div>
          ))}
        </div>

        {/* Form Card */}
        <Card className="border-0 shadow-2xl bg-white/80 backdrop-blur-xl">
          <CardContent className="p-8">
            {/* Slide Header */}
            <div className="flex items-center gap-4 mb-8 pb-6 border-b border-gray-200">
              <div className="w-14 h-14 bg-gradient-to-br from-emerald-100 to-teal-100 rounded-2xl flex items-center justify-center">
                <SlideIcon className="w-7 h-7 text-emerald-600" />
              </div>
              <div>
                <h2 className="text-2xl font-bold text-gray-900">{currentSlideInfo.title}</h2>
                <p className="text-gray-600">{currentSlideInfo.subtitle}</p>
              </div>
            </div>

            {/* Slide 0: Account Credentials */}
            {currentSlide === 0 && (
              <div className="space-y-6 animate-in fade-in duration-500">
                <div className="space-y-2">
                  <Label htmlFor="email" className="text-gray-700 font-medium flex items-center gap-2">
                    Email Address <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <div className="relative">
                    <Mail className="absolute left-3 top-1/2 transform -translate-y-1/2 w-5 h-5 text-gray-400" />
                    <Input
                      type="email"
                      id="email"
                      name="email"
                      value={formData.email}
                      onChange={handleInputChange}
                      className="pl-10 h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="Enter your email"
                      required
                    />
                  </div>
                </div>

                <div className="space-y-2">
                  <Label htmlFor="username" className="text-gray-700 font-medium flex items-center gap-2">
                    Username <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <div className="relative">
                    <User className="absolute left-3 top-1/2 transform -translate-y-1/2 w-5 h-5 text-gray-400" />
                    <Input
                      type="text"
                      id="username"
                      name="username"
                      value={formData.username}
                      onChange={handleInputChange}
                      className="pl-10 h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="Choose a username"
                      required
                    />
                  </div>
                  <p className="text-xs text-gray-500">Min 3 characters, no spaces</p>
                </div>

                <div className="space-y-2">
                  <Label htmlFor="password" className="text-gray-700 font-medium flex items-center gap-2">
                    Password <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <div className="relative">
                    <Lock className="absolute left-3 top-1/2 transform -translate-y-1/2 w-5 h-5 text-gray-400" />
                    <Input
                      type={showPassword ? 'text' : 'password'}
                      id="password"
                      name="password"
                      value={formData.password}
                      onChange={handleInputChange}
                      className="pl-10 pr-12 h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="Create a password"
                      required
                    />
                    <button
                      type="button"
                      onClick={() => setShowPassword(!showPassword)}
                      className="absolute right-3 top-1/2 transform -translate-y-1/2 text-gray-400 hover:text-gray-600"
                    >
                      {showPassword ? <EyeOff className="w-5 h-5" /> : <Eye className="w-5 h-5" />}
                    </button>
                  </div>
                  <p className="text-xs text-gray-500">Min 6 characters</p>
                </div>

                <div className="space-y-2">
                  <Label htmlFor="confirmPassword" className="text-gray-700 font-medium flex items-center gap-2">
                    Confirm Password <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <div className="relative">
                    <Lock className="absolute left-3 top-1/2 transform -translate-y-1/2 w-5 h-5 text-gray-400" />
                    <Input
                      type={showConfirmPassword ? 'text' : 'password'}
                      id="confirmPassword"
                      name="confirmPassword"
                      value={formData.confirmPassword}
                      onChange={handleInputChange}
                      className="pl-10 pr-12 h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="Confirm your password"
                      required
                    />
                    <button
                      type="button"
                      onClick={() => setShowConfirmPassword(!showConfirmPassword)}
                      className="absolute right-3 top-1/2 transform -translate-y-1/2 text-gray-400 hover:text-gray-600"
                    >
                      {showConfirmPassword ? <EyeOff className="w-5 h-5" /> : <Eye className="w-5 h-5" />}
                    </button>
                  </div>
                </div>
              </div>
            )}

            {/* Slide 1: Basic Info */}
            {currentSlide === 1 && (
              <div className="space-y-6 animate-in fade-in duration-500">
                <div className="space-y-2">
                  <Label htmlFor="name" className="text-gray-700 font-medium flex items-center gap-2">
                    Full Name <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <Input
                    type="text"
                    id="name"
                    name="name"
                    value={formData.name}
                    onChange={handleInputChange}
                    className="h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                    placeholder="Enter your full name"
                    required
                  />
                </div>

                {/* <div className="space-y-3">
                  <Label className="text-gray-700 font-medium flex items-center gap-2">
                    Gender <Badge variant="destructive" className="text-xs">Required</Badge>
                  </Label>
                  <div className="grid grid-cols-3 gap-3">
                    {GENDER_OPTIONS.map((gender) => (
                      <button
                        key={gender}
                        type="button"
                        onClick={() => setFormData({ ...formData, gender })}
                        className={`py-4 rounded-xl font-medium transition-all duration-300 ${
                          formData.gender === gender
                            ? 'bg-gradient-to-r from-emerald-600 to-teal-600 text-white shadow-lg scale-105'
                            : 'bg-gray-100 text-gray-700 hover:bg-gray-200 hover:scale-105'
                        }`}
                      >
                        {gender}
                      </button>
                    ))}
                  </div>
                </div>

                <div className="space-y-2">
                  <Label htmlFor="age" className="text-gray-700 font-medium flex items-center gap-2">
                    Age <Badge variant="secondary" className="text-xs">Optional</Badge>
                  </Label>
                  <Input
                    type="number"
                    id="age"
                    name="age"
                    value={formData.age}
                    onChange={handleInputChange}
                    className="h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                    placeholder="Enter your age"
                    min="1"
                    max="120"
                  />
                </div> */}
              </div>
            )}

            {/* Slide 2: Dietary Preferences
            {currentSlide === 2 && (
              <div className="space-y-6 animate-in fade-in duration-500">
                <div className="flex justify-between items-center">
                  <Badge variant="secondary" className="text-xs">Optional - You can skip</Badge>
                </div>

                <div className="space-y-3">
                  <Label className="text-gray-700 font-medium">Choose Your Diet Preference</Label>
                  <div className="grid grid-cols-1 sm:grid-cols-3 gap-3">
                    {DIET_OPTIONS.map((diet) => (
                      <button
                        key={diet}
                        type="button"
                        onClick={() => handleDietSelect(diet)}
                        className={`py-4 px-4 rounded-xl font-medium transition-all duration-300 ${
                          formData.dietPreference === diet
                            ? 'bg-gradient-to-r from-emerald-600 to-teal-600 text-white shadow-lg scale-105'
                            : 'bg-gray-100 text-gray-700 hover:bg-gray-200 hover:scale-105'
                        }`}
                      >
                        {diet}
                      </button>
                    ))}
                  </div>
                </div>

                <div className="space-y-3">
                  <Label className="text-gray-700 font-medium">Select Your Allergens (if any)</Label>
                  <div className="grid grid-cols-2 sm:grid-cols-4 gap-3">
                    {ALLERGEN_OPTIONS.map((allergen) => (
                      <button
                        key={allergen}
                        type="button"
                        onClick={() => handleAllergenToggle(allergen)}
                        className={`py-3 px-3 rounded-xl text-sm font-medium transition-all duration-300 ${
                          formData.allergens.includes(allergen)
                            ? 'bg-gradient-to-r from-rose-500 to-red-600 text-white shadow-lg scale-105'
                            : 'bg-gray-100 text-gray-700 hover:bg-gray-200 hover:scale-105'
                        }`}
                      >
                        {allergen}
                      </button>
                    ))}
                  </div>
                </div>
              </div>
            )} */}
{/* 
            Slide 3: Health Focus
            {currentSlide === 3 && (
              <div className="space-y-6 animate-in fade-in duration-500">
                <div className="flex justify-between items-center">
                  <Badge variant="secondary" className="text-xs">Optional - You can skip</Badge>
                </div>

                <div className="space-y-3">
                  <Label className="text-gray-700 font-medium">Choose Your Health Focus</Label>
                  <div className="grid grid-cols-1 sm:grid-cols-3 gap-3">
                    {HEALTH_FOCUS_OPTIONS.map((focus) => (
                      <button
                        key={focus}
                        type="button"
                        onClick={() => handleHealthFocusToggle(focus)}
                        className={`py-4 px-4 rounded-xl font-medium transition-all duration-300 ${
                          formData.healthFocus.includes(focus)
                            ? 'bg-gradient-to-r from-blue-600 to-cyan-600 text-white shadow-lg scale-105'
                            : 'bg-gray-100 text-gray-700 hover:bg-gray-200 hover:scale-105'
                        }`}
                      >
                        {focus}
                      </button>
                    ))}
                  </div>
                </div>

                <div className="grid grid-cols-2 gap-4">
                  <div className="space-y-2">
                    <Label htmlFor="height" className="text-gray-700 font-medium flex items-center gap-2">
                      <Ruler className="w-4 h-4" />
                      Height (cm)
                    </Label>
                    <Input
                      type="number"
                      id="height"
                      name="height"
                      value={formData.height}
                      onChange={handleInputChange}
                      className="h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="e.g., 170"
                      min="1"
                    />
                  </div>

                  <div className="space-y-2">
                    <Label htmlFor="weight" className="text-gray-700 font-medium flex items-center gap-2">
                      <Weight className="w-4 h-4" />
                      Weight (kg)
                    </Label>
                    <Input
                      type="number"
                      id="weight"
                      name="weight"
                      value={formData.weight}
                      onChange={handleInputChange}
                      className="h-12 border-gray-300 focus:border-emerald-500 focus:ring-emerald-500"
                      placeholder="e.g., 70"
                      min="1"
                      step="0.1"
                    />
                  </div>
                </div>
              </div>
            )} */}
                      

            {/* Error Message */}
            {error && (
              <Alert variant="destructive" className="mt-4">
                <AlertDescription>{error}</AlertDescription>
              </Alert>
            )}

            {/* Navigation Buttons */}
            <div className="flex justify-between items-center mt-8 pt-6 border-t border-gray-200 gap-4">
              <Button
                onClick={handlePrevious}
                disabled={currentSlide === 0}
                variant="outline"
                className="px-6 h-12"
              >
                <ChevronLeft className="w-5 h-5 mr-2" />
                Previous
              </Button>

              <div className="flex gap-3">
                {currentSlide > 1 && (
                  <Button
                    onClick={handleSkip}
                    variant="ghost"
                    className="text-gray-600 hover:text-gray-800"
                  >
                    Skip
                  </Button>
                )}

                {currentSlide < 1 ? (
                  <Button
                    onClick={handleNext}
                    className="px-6 h-12 bg-gradient-to-r from-emerald-600 to-teal-600 hover:from-emerald-700 hover:to-teal-700 text-white shadow-lg hover:shadow-xl hover:shadow-emerald-500/50"
                  >
                    Next
                    <ChevronRight className="w-5 h-5 ml-2" />
                  </Button>
                ) : (
                  <Button
                    onClick={handleSubmit}
                    className="px-8 h-12 bg-gradient-to-r from-emerald-600 to-teal-600 hover:from-emerald-700 hover:to-teal-700 text-white shadow-lg hover:shadow-xl hover:shadow-emerald-500/50"
                  >
                    <Sparkles className="w-5 h-5 mr-2" />
                    Complete Setup
                  </Button>
                )}
              </div>
            </div>
          </CardContent>
        </Card>

        {/* Sign In Link */}
        <div className="text-center mt-6">
          <p className="text-gray-600">
            Already have an account?{' '}
            <button
              onClick={() => navigate('/login')}
              className="text-emerald-600 hover:text-emerald-700 font-semibold transition-colors"
            >
              Sign In
            </button>
          </p>
        </div>

        {/* Back to Home */}
        <div className="text-center mt-4">
          <button
            onClick={() => navigate('/')}
            className="text-gray-600 hover:text-emerald-600 font-medium transition-colors"
          >
            ← Back to Home
          </button>
        </div>
      </div>
    </div>
  );
};

export default Signup;
